blocks/file-blocks/sandbox/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { tw } from "twind";
+import { ReactNode, useEffect, useState } from "react";
+import LZString from "lz-string";
+import { FileBlockProps } from "@githubnext/blocks";
+
+const optionsDefaults = {
+  fontsize: 14,
+  hidenavigation: 1,
+  codemirror: 1,
+  hidedevtools: 0,
+  expanddevtools: 1,
+  view: "split",
+};
+export default ({ content, state = {} }: FileBlockProps) => {
+  const [url, setUrl] = useState("");
+  const parameters = getParameters({
+    files: {
+      "index.js": {
+        // @ts-ignore
+        content,
+        isBinary: false,
+      },
+      "package.json": {
+        content: JSON.stringify({
+          dependencies: getDependenciesFromString(content),
+        }),
+        isBinary: false,
+      },
+    },
+  });
+
+  const getSandboxUrl = async () => {
+    const url = `https://codesandbox.io/api/v1/sandboxes/define?parameters=${parameters}&json=1`;
+    const res = await fetch(url);
+    const data = await res.json();
+    const id = data?.sandbox_id;
+    const params = new URLSearchParams({
+      ...optionsDefaults,
+      ...(state.sandboxOptions || {}),
+    }).toString();
+    const iframeUrl = `https://codesandbox.io/embed/${id}?${params}`;
+
+    setUrl(iframeUrl);
+  };
+  useEffect(() => {
+    getSandboxUrl();
+  }, []);
```

Comment:
I wonder if we can eventually refactor here to use `React.Suspense` so that we can [render as we fetch](https://17.reactjs.org/docs/concurrent-mode-suspense.html).